internal/testutil/reindent_test.go

```diff
@@ -70,3 +70,34 @@ func (*S) testReindent(c *C, test reindentTest) {
 	}
 	c.Assert(string(reindented), Equals, test.result)
 }
+
+type prefixEachLineTest struct {
+	raw, prefix, result string
+}
+
+var prefixEachLineTests = []prefixEachLineTest{{
+	raw:    "a\n\tb\n  \t\tc\td\n\t ",
+	prefix: "foo",
+	result: "fooa\nfoo\tb\nfoo  \t\tc\td\nfoo\t ",
+}, {
+	raw:    "foo",
+	prefix: "pref",
+	result: "preffoo",
+}, {
+	raw:    "",
+	prefix: "",
+	result: "",
+}, {
+	raw:    "\n",
+	prefix: "\t",
+	result: "\t\n\t",
```

Comment:
This doesn't look right. It's indenting an empty line with zero content, and may create actual issues depending on what else shows up after it. Imagine AddPrefix("foo\n") + "bar".

Reply:
I agree with you. We went for a naive implementation which produces strange cases. I have changed it and made it more complex.